src/network/networkd-ndisc.c

```diff
@@ -951,7 +952,16 @@ static int ndisc_router_process_options(Link *link, sd_ndisc_router *rt) {
                         r = ndisc_router_process_dnssl(link, rt);
                         break;
                 case SD_NDISC_OPTION_CAPTIVE_PORTAL:
+                        if (n_captive_portal > 0) {
+                                if (n_captive_portal == 1)
+                                        log_link_debug(link, "Received RA with multiple captive portals, only using the first one.");
```

Comment:
>  if this actually occurs in the wild. Either it doesn't at all, and there is no issue

The issue here is that it's possible to tirgger OOMs in networkd and crash it by sending router advertisements with a gazillion different captive portals.

Reply:
Just to clarify I agree it should be `log_info` because if there are RAs with more than one captive portal it means that something fishy is going on and it should be logged somewhere without having to set the log level to debug. Then again, I still don't think all the sets is needed here.

Reply:
Yes, that's what I meant too.

Reply:
> > if this actually occurs in the wild. Either it doesn't at all, and there is no issue
> 
> The issue here is that it's possible to tirgger OOMs in networkd and crash it by sending router advertisements with a gazillion different captive portals.

Now this PR sets upper limit, then OOM will not be triggered.